src/System.Binary.Base64/System/Binary/Base64.cs

```diff
@@ -135,25 +142,37 @@ public static int Encode(ReadOnlySpan<byte> source, Span<byte> destination)
             while (sourceIndex < srcLength - 2)
             {
                 result = Encode(ref Unsafe.Add(ref srcBytes, sourceIndex));
+                if (destIndex > destLength - 4) goto FalseExit;
```

Comment:
Does it hurt performance to do this check before we read the source and compute the destination bytes? If not, I'd move it since you are doing work that would be throwaway if the destination buffer doesn't have room.